ansible_builder/requirements.py

```diff
@@ -37,6 +38,7 @@ def sanitize_requirements(collection_py_reqs):
     for collection, lines in collection_py_reqs.items():
         try:
             for req in requirements.parse('\n'.join(lines)):
+                req.name = safe_name(req.name)
```

Comment:
```suggestion
                if req.specifier:
                    req.name = safe_name(req.name)
```

Looks like `specifier` gets set to true only in the simple package path.  Would avoid normalizing non-package names.